clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -652,8 +652,13 @@ public synchronized void resume(TopicPartition tp) {
     }
 
     synchronized void requestFailed(Set<TopicPartition> partitions, long nextRetryTimeMs) {
-        for (TopicPartition partition : partitions)
-            assignedState(partition).requestFailed(nextRetryTimeMs);
+        for (TopicPartition partition : partitions) {
```

Comment:
This is another orthogonal minor bug I found from the flaky tests: `requestFailed` can be called in the response handler, in which the subscription may has been changed already.